src/main/java/org/mariadb/jdbc/internal/osgi/MariaDbDataSourceFactory.java

```diff
@@ -0,0 +1,134 @@
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with this library; if not, write to Monty Program Ab info@montyprogram.com.
+ *
+ * This particular MariaDB Client for Java file is work
+ * derived from a Drizzle-JDBC. Drizzle-JDBC file which is covered by subject to
+ * the following copyright and notice provisions:
+ *
+ * Copyright (c) 2009-2011, Marcus Eriksson
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list
+ * of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice, this
+ * list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.
+ *
+ * Neither the name of the driver nor the names of its contributors may not be
+ * used to endorse or promote products derived from this software without specific
+ * prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS  AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+ * OF SUCH DAMAGE.
+ *
+ */
+
+package org.mariadb.jdbc.internal.osgi;
+
+import java.sql.SQLException;
+import java.util.Properties;
+
+import javax.sql.ConnectionPoolDataSource;
+import javax.sql.XADataSource;
+
+import org.mariadb.jdbc.Driver;
+import org.mariadb.jdbc.MariaDbDataSource;
+import org.osgi.service.jdbc.DataSourceFactory;
+
+public class MariaDbDataSourceFactory implements DataSourceFactory {
+
+  @Override
+  public MariaDbDataSource createDataSource(Properties props) throws SQLException {
+    MariaDbDataSource source = new MariaDbDataSource();
```

Comment:
Implementation must distinguished basic implementation (MariaDbDataSource) and Connection pooling implementation (MariaDbPoolDataSource) if according properties JDBC_INITIAL_POOL_SIZE, JDBC_MAX_IDLE_TIME, JDBC_MAX_POOL_SIZE, JDBC_MAX_STATEMENTS, JDBC_MIN_POOL_SIZE are set

JDBC_PROPERTY_CYCLE

Reply:
One Problem I noticed: MariaDbPoolDataSource does not implements ConnectionPoolDataSource but DataSourceFactory#createConnectionPoolDataSource needs one.
On the other hand MariaDbDataSource DOES implement ConnectionPoolDataSource even though it seems it is not really one.

Reply:
Maybe it was meant that MariaDbPoolDataSource extends MariaDbDataSource and implements ConnectionPoolDataSource?

Reply:
Right, that has to be corrected in 2.5. Issue created: https://jira.mariadb.org/browse/CONJ-692

Reply:
So would it be possible to use the implementation as-is (even if not 100% correct in the sense of connection pooling) and change it when CONJ-692 is fixed?